docs/developers/nft-tokens.md

```diff
@@ -396,6 +408,51 @@ StopNFTCreationTransaction {
 }
 ```
 
+### **Change NFT Attributes**
+
+A user that has the `ESDTRoleNFTUpdateAttributes` role set for a given ESDT, can change the attributes of a given NFT/SFT.
+
+:::tip
+`ESDTNFTUpdateAttributes` will remove the old attributes and add the new ones. Therefore, if you want to keep the old attributes you will have to pass them along with the new ones.
+:::
+This is done by performing a transaction like this:
+
+```
+ESDTNFTUpdateAttributesTransaction {
+    Sender: <address of an address that has ESDTRoleNFTUpdateAttributes role>
+    Receiver: <same as sender>
+    Value: 0
+    GasLimit: 10000000
+    Data: "ESDTNFTUpdateAttributes" +
+          "@" + <token identifier in hexadecimal encoding> +
+          "@" + <NFT nonce in hexadecimal encoding> +
+          "@" + <Attributes in hexadecimal encoding>
+}
+```
+To see how you can assign this role in case it is not set, please refer to [this](/developers/nft-tokens#assigning-roles) section.
+
+
+### **Add URIs to NFT**
+
+A user that has the `ESDTRoleNFTAddURI` role set for a given ESDT, can add uris to a given NFT/SFT.
```

Comment:
Same story as above.

Reply:
Any timeline for adding these to SC in rust?
I tried to add it myself inside the ESDTSystemSmartContractProxy, but I got some errors.
the method update_attributes exists for struct ESDTSystemSmartContractProxy<<Self as ContractBase>::Api>, but its trait bounds were not satisfied


```
pub fn update_attributes(
        self,
        token_identifier: &TokenIdentifier<SA>,
        nonce: u64,
        attributes: &ManagedBuffer<SA>
    ) -> ContractCall<SA, ()> {
        let mut contract_call = self.esdt_system_sc_call_no_args(b"updateAttributes");

        contract_call.push_endpoint_arg(token_identifier);
        contract_call.push_endpoint_arg(nonce);
        contract_call.push_endpoint_arg(attributes);

        contract_call
    }

```

Reply:
Can't imagine what the error could be. Could you copy&paste it here? The function should look like the `mint` function from the same file (system_sc_proxy.rs)

Reply:
The calling part looks something like this:

```
#[endpoint(updateAttributes)]
    fn update_attributes(&self, nonce: u64, attributes: ManagedBuffer) -> SCResult<AsyncCall> {
        Ok(self
            .send()
            .esdt_system_sc_proxy()
            .update_attributes(
                &self.nft_token_id().get(),
                nonce,
                &attributes
            )
            .async_call())
    }
```

Reply:
I've just tried almost the exact same thing and worked :thinking: 

```
    pub fn update_attributes(
        self,
        token_identifier: &TokenIdentifier<SA>,
        nonce: u64,
        attributes: &ManagedBuffer<SA>,
    ) -> ContractCall<SA, ()> {
        let mut contract_call = self.esdt_system_sc_call_no_args(b"updateAttributes");

        contract_call.push_endpoint_arg(token_identifier);
        contract_call.push_endpoint_arg(nonce);
        contract_call.push_endpoint_arg(attributes);

        contract_call
    }
```

```
    #[endpoint(updateAttributes)]
    fn update_attributes(
        &self,
        token: TokenIdentifier,
        nonce: u64,
        attributes: ManagedBuffer,
    ) -> SCResult<AsyncCall> {
        Ok(self
            .send()
            .esdt_system_sc_proxy()
            .update_attributes(&token, nonce, &attributes)
            .async_call())
    }
```

Reply:
Strange. I'll triple check then. Thanks a lot for checking!